AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2188,14 +2228,17 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
     ASEnvironmentStatePropagateDown(layoutSpec, self.environmentTraitCollection);
     
     layoutSpec.isMutable = NO;
-    ASLayout *layout = [layoutSpec measureWithSizeRange:constrainedSize];
+    ASLayout *layout = [layoutSpec calculateLayoutThatFits:constrainedSize parentSize:constrainedSize.max];
     ASDisplayNodeAssertNotNil(layout, @"[ASLayoutSpec measureWithSizeRange:] should never return nil! %@, %@", self, layoutSpec);
       
     // Make sure layoutableObject of the root layout is `self`, so that the flattened layout will be structurally correct.
     BOOL isFinalLayoutable = (layout.layoutableObject != self);
     if (isFinalLayoutable) {
       layout.position = CGPointZero;
-      layout = [ASLayout layoutWithLayoutableObject:self constrainedSizeRange:constrainedSize size:layout.size sublayouts:@[layout]];
+      layout = [ASLayout layoutWithLayoutableObject:self
```

Comment:
This may have been decided previously, but is it necessary to have `layoutWithLayoutableObject:constrainedSize:etc:`? Why not just `layoutWith:constrainedSize:etc:`?